src/expr/src/relation/mod.rs

```diff
@@ -654,6 +692,27 @@ impl MirRelationExpr {
         }
     }
 
+    /// The number of child relations this relation has.
+    pub fn num_inputs(&self) -> usize {
+        match self {
```

Comment:
Can you add a unit test to the bottom of this file to confirm that `num_inputs` is always equal to the result of:
```
let mut count = 0 
self.visit1(|_| count += 1)
count
```

Reply:
Could potentially also replace the function with this; I think it should optimize down to probably the same thing.

Reply:
Good point. I'll do it and measure it

Reply:
It does indeed optimize down to the same thing. Replaced. Thanks!